tests/db_functions/datetime/test_extract_trunc.py

```diff
@@ -1935,3 +1935,113 @@ def test_trunc_invalid_field_with_timezone(self):
+                        ),
+                    ),
+                    (
+                        end_datetime,
+                        truncate_to(
+                            end_datetime.astimezone(etc_timezone), kind, etc_timezone
+                        ),
+                    ),
+                ],
+                lambda m: (m.start_datetime, m.truncated),
+            )
+
+        def test_datetime_to_date_kind(kind):
+            self.assertQuerySetEqual(
+                DTModel.objects.annotate(
+                    truncated=Trunc(
+                        "start_datetime",
+                        kind,
+                        output_field=DateField(),
+                        tzinfo=etc_timezone,
+                    ),
+                ).order_by("start_datetime"),
+                [
+                    (
+                        start_datetime,
+                        truncate_to(
+                            start_datetime.astimezone(etc_timezone).date(), kind
+                        ),
+                    ),
+                    (
+                        end_datetime,
+                        truncate_to(end_datetime.astimezone(etc_timezone).date(), kind),
+                    ),
+                ],
+                lambda m: (m.start_datetime, m.truncated),
+            )
+
+        def test_datetime_to_time_kind(kind):
+            self.assertQuerySetEqual(
+                DTModel.objects.annotate(
+                    truncated=Trunc(
+                        "start_datetime",
+                        kind,
+                        output_field=TimeField(),
+                        tzinfo=etc_timezone,
+                    )
+                ).order_by("start_datetime"),
+                [
+                    (
+                        start_datetime,
+                        truncate_to(
+                            start_datetime.astimezone(etc_timezone).time(), kind
+                        ),
+                    ),
+                    (
+                        end_datetime,
+                        truncate_to(end_datetime.astimezone(etc_timezone).time(), kind),
+                    ),
+                ],
+                lambda m: (m.start_datetime, m.truncated),
+            )
+
+        test_datetime_to_date_kind("year")
+        test_datetime_to_date_kind("quarter")
+        test_datetime_to_date_kind("month")
+        test_datetime_to_date_kind("week")
+        test_datetime_to_date_kind("day")
+        test_datetime_to_time_kind("hour")
+        test_datetime_to_time_kind("minute")
+        test_datetime_to_time_kind("second")
+        test_datetime_kind("year")
+        test_datetime_kind("quarter")
+        test_datetime_kind("month")
+        test_datetime_kind("week")
+        test_datetime_kind("day")
+        test_datetime_kind("hour")
+        test_datetime_kind("minute")
+        test_datetime_kind("second")
```

Comment:
```suggestion
        assert_datetime_to_date_kind("year")
        assert_datetime_to_date_kind("quarter")
        assert_datetime_to_date_kind("month")
        assert_datetime_to_date_kind("week")
        assert_datetime_to_date_kind("day")
        assert_datetime_to_time_kind("hour")
        assert_datetime_to_time_kind("minute")
        assert_datetime_to_time_kind("second")
        assert_datetime_kind("year")
        assert_datetime_kind("quarter")
        assert_datetime_kind("month")
        assert_datetime_kind("week")
        assert_datetime_kind("day")
        assert_datetime_kind("hour")
        assert_datetime_kind("minute")
        assert_datetime_kind("second")
```

Reply:
This seems like a good opportunity to use [subtests](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) to me.